server/src/main/java/org/opensearch/rest/action/document/RestDeleteAction.java

```diff
@@ -56,13 +56,7 @@
 
     @Override
     public List<Route> routes() {
-        return unmodifiableList(
-            asList(
-                new Route(DELETE, "/{index}/_doc/{id}"),
-                // Deprecated typed endpoint.
-                new Route(DELETE, "/{index}/{type}/{id}")
-            )
-        );
+        return unmodifiableList(asList(new Route(DELETE, "/{index}/_doc/{id}")));
```

Comment:
Shouldn't it be just `"/{index}/{id}"`? Afaik, the mapping might be changed from `_doc`, but the key concept - only single mapping allowed - will stay. May be event better to support both:

```
asList(
                new Route(DELETE, "/{index}/_doc/{id}"),
                new Route(DELETE, "/{index}/{id}")
            )
```

Reply:
Thanks for taking time and reviewing this PR.  

Yes, keeping both suggested end-points make sense. 
The only point I have it against is document index APIs have either `_doc` or `_create` in the path where it provides document index behaviour instead of document type. 

But, I think for delete action, we don't need this segregation and we can use both end-points as you suggested.

Reply:
:+1: makes sense, thanks!